src/Main.elm

```diff
@@ -46,6 +51,15 @@ init =
         watchDate movie =
             Movie.watchDate movie
                 |> Maybe.withDefault (date 1800 1 1)
+
+        selectOptions set =
+            Set.map (\m -> ( m, m )) set
+                |> Set.toList
+
+        genres =
```

Comment:
Using the Set of genres on each movie, use Foldr to create a new union Set, which is all unique genres.